### PR TITLE
Invitations

### PR DESCRIPTION
**Pull request by elguapogordo**
added fixtures for test stubbing
added test, user can request an invitation email for a github user with a public email address
added and updated routes
added form for github user name
added mailer controller
added function for user github name to be included in invitation emails
fixed some existing typos
placated rubocop
added test, admin can add video with happy path and sad path
added test, function, videos have validations now
added test, homepage links to about page
added link to about page and fixed spacing

rubocop.happy? => true
all_tests_passing? => true

### CODE REVIEW

File: app/controllers/invites_controller.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  def send_invite
+    InviteMailer.invite(gather_email_params).deliver_now
+    flash[:success] = 'Successfully sent invite!'
+  end
+
+  def no_email
+    message1 = "The Github user you selected doesn't have an "
+    message2 = 'email address associated with their account.'
+    flash[:notice] = "#{message1}#{message2}"
+  end
```
Comment (elguapogordo) [inline app/controllers/invites_controller.rb:32]: i don't like it this way but rubocop does not tolerate long strings

File: app/models/video.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class Video < ApplicationRecord
+  validates :title, presence: true
+  validates :description, presence: true
+  validates :video_id, presence: true
```
Comment (elguapogordo) [inline app/models/video.rb:4]: i'm a bit surprised they didn't have validations before

File: app/views/invite_mailer/invite.html.erb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Hello <%= @name %>,
+
+<%= @sender_name %> has invited you to join Brownfield of Dreams. You can create an account <%= link_to 'here', signup_url %>.
```
Comment (elguapogordo) [inline app/views/invite_mailer/invite.html.erb:3]: `@sender_name` verified in pry. it works!

File: app/views/partials/_nav.html.erb [hunk 4]
```diff
@@ -18,8 +18,9 @@
       <% end %>
     <% else %>
       <div class="align-middle table-cell p3">
-        <%= link_to 'Sign In', login_path, class: "px2"%>
-        <%= link_to 'Register', register_path %>
+        <%= link_to 'Sign In', login_path, class: "px2" %>
+        <%= link_to 'Register', register_path, class: "px2" %>
+        <%= link_to 'About', about_path, class: "px2" %>
```
Comment (elguapogordo) [inline app/views/partials/_nav.html.erb:23]: `class: "px2"` on only one link made the spacing asymmetrical, so i gave them all spacing.

File: app/controllers/invites_controller.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
+class InvitesController < ApplicationController
+  def new; end
+
+  def create
+    invitee = gather_email_params
+    invitee[:email] ? send_invite : no_email
```
Comment (cmpprg) [inline app/controllers/invites_controller.rb:6]: looks clean

File: app/controllers/invites_controller.rb (hunk 1 above)
Comment (cmpprg) [reply app/controllers/invites_controller.rb:32]: RuboCop!!!!!

File: app/views/invite_mailer/invite.html.erb (hunk 3 above)
Comment (cmpprg) [reply app/views/invite_mailer/invite.html.erb:3]: Sounds good

File: app/views/partials/_nav.html.erb [hunk 6]
```diff
@@ -18,8 +18,9 @@
       <% end %>
     <% else %>
       <div class="align-middle table-cell p3">
-        <%= link_to 'Sign In', login_path, class: "px2"%>
-        <%= link_to 'Register', register_path %>
+        <%= link_to 'Sign In', login_path, class: "px2" %>
```
Comment (cmpprg) [inline app/views/partials/_nav.html.erb:21]: is sign in different than register?

**Review by cmpprg — APPROVED** [review]
This look pretty good! I approve, I will go ahead and merge

**Pull request merged**